### ISSUE TITLE
Set up failover

### ISSUE THREAD

**Issue by finestructure** [SwiftPackageIndex/SwiftPackageIndex-Server#722]
We should set up a way to be able to fail over to staging in case something goes wrong with prod. Every time we're doing a system update I'm a bit worried that it might get stuck or take ages. It would be good if we failed over to staging to give us time and sort things out without downtime for users.

- [ ] document where to change the IP forwarding and ensure we both have access
- [x] add env variable to temporarily suppress staging/development banner

**Comment by finestructure** [comment]
I fear exactly this may be happening right now. I triggered the 10.15.7 update via the command line and when it didn't progress after an hour I triggered rebooted. prod's been down for ~10min now :/

**Comment by finestructure** [comment]
prod doesn't seem to be coming back up by itself. I'm hesitating to do a hard reset - I don't want to make things worse by cutting power while it's in an unknown state.

**Comment by daveverwer** [comment]
I'm guessing you're not getting notifications. I'm messaging you, and you're clearly there but I don't think they're getting through.

**Comment by finestructure** [comment]
And just as I wrote this it's back up. Still, we need to sort out the failover. I'd like to be able to run system updates without downtime.

**Comment by daveverwer** [comment]
That's great news. I'd like to talk to you about the failover though, check your Discord 👍

**Comment by finestructure** [comment]
After yesterday's outage I think it's time to look at this again. I think we might get a lot of mileage out of letting Cloudflare cache everything: https://support.cloudflare.com/hc/en-us/articles/202775670-How-do-I-cache-static-HTML-

It might not even require a pro account. The only thing to check is if there's a way to have it use the cache instead of throwing a 52x while also not having a TTL that's interfering with our analysis refresh. I.e. while most of our content is fairly static we don't want to add too much / another delay until updates are shown.

**Comment by finestructure** [comment]
See also: https://support.cloudflare.com/hc/en-us/articles/200168436

**Comment by finestructure** [comment]
I have this setting for https://finestructure.co (I'm on a free CF account)

<img width="887" alt="Screenshot 2020-10-21 at 08 34 57" src="https://user-images.githubusercontent.com/65520/96681924-4e901200-1378-11eb-8849-b104271f264a.png">

Do we not have this on for the SPI? If we've got this enabled and we're seeing 52x then it's not working as intended.

**Comment by daveverwer** [comment]
Yes, that setting is already on.

**Comment by finestructure** [comment]
Well, that's concerning. Why are we showing 521s then when prod is down? My reading of the docs is that it should prevent exactly that from happening?

**Comment by daveverwer** [comment]
My reading of that setting makes it seem a bit flakey at best, if it's using Wayback Machine snapshots as a source, are we even listed there yet? Certainly, that data isn't going to be particularly up to date.

Yesterday's issue aside, I don't worry about a few minutes of downtime here and there for system updates. If the server starts crashing all the time and we have hours of downtime a week, I think we need to look into it more seriously, but for things like system updates I think we could add a lot of complexity for very real real-world gain.

**Comment by finestructure** [comment]
My understanding is that wayback is sort of optional. But even so - that's still better than the 521 page I think.

The downtime yesterday was more than 1.5 hours. But the bigger problem was that for a significant amount of time I was uncertain whether we'd have the service back up until we get hands on from someone at MacStadium.

We've had two warning shots already. I just don't like not even having a fall back plan... :( If it hadn't come back up by itself last night, we'd probably still be down this morning and it's not clear how we could/would restore service.

**Comment by finestructure** [comment]
We’ve agreed to fail over to staging (by changing the IP) and to disable the banner via the CI variable `HIDE_STAGING_BANNER` on staging/dev.

I don’t think it’s possible for both of us to have access to the IP redirect for now. Leaving this open for now to decide if we need to do anything further or if we just leave it at that.

**Comment by daveverwer** [comment]
Inspired by this, I decided to go and create a new CloudFlare account so we could both have access, but I'd also have to make a new email account.

![Screenshot 2020-11-01 at 11 28 19@2x](https://user-images.githubusercontent.com/5180/97801562-93982c00-1c35-11eb-8214-9bf75004de89.png)

We can do that, if we decide it's worth it. However, I'm rarely very far from my phone where I can do a failover if necessary. Let's re-open this after we have a catastrophic downtime where I just got on a 12-hour plane journey 😂
